### PR TITLE
TINY-9560: Fix `color_cols` option not being respected

### PR DESCRIPTION
**Pull request by MitchCTiny**
Related Ticket: TINY-9560

Description of Changes:
* `forecolor` and `backcolor` color swatches did not have the correct number of columns when `color_cols` option was set
* `color_cols` option is now used as a fallback before the default number of columns for the `forecolor` and `backcolor` color swatches

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by vpyshnenko — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts [hunk 1]
```diff
@@ -92,12 +97,12 @@ const register = (editor: Editor): void => {
 
   registerOption('color_cols_foreground', {
     processor: 'number',
-    default: calcCols(getColors(editor, foregroundId).length)
+    default: calcColsOption(calcCols(getColors(editor, backgroundId).length), option('color_cols')(editor))
```
[the remark this answers is not in the document]
Comment (MitchCTiny) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts:100]: Must've missed that, thanks for catching it.

[the remark this answers is not in the document]
Comment (spocke) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts]: Why didn't tests catch this? Are we missing one?

[the remark this answers is not in the document]
Comment (MitchCTiny) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts]: Just added tests for this now. There wasn't a test for the calculated column numbers when taken from the new color map options, only from specified column numbers.

**Review by TheSpyder — APPROVED** [review]
Since we repeat `option('color_cols')(editor)` three times now I wonder if it would be better refactored somehow? Otherwise looks good

**Pull request merged**